models/gmx/arbitrum/glp/gmx_arbitrum_glp_fees.sql

```diff
@@ -21,53 +21,58 @@ GLP fees accrued to the Fee GLP contract and can be claimed by invoking function
-            FROM
-                (
-                SELECT -- This subquery truncates the block time to a minute and selects all the inbound tranfers of WETH tokens to the Fee GLP contract
-                    date_trunc('minute', evt_block_time) AS minute,
-                    ((value) / 1e18) AS transfer_value -- WETH 18dp
-                FROM {{ source('erc20_arbitrum', 'evt_transfer') }}
-                WHERE `to` = '0x4e971a87900b931ff39d1aad67697f49835400b6' -- Fee GLP contract
-                AND `contract_address` = '0x82af49447d8a07e3bd95bd0d56f35241523fbab1' -- WETH Arbitrum Smart Contract
-                ) a
-            GROUP BY a.minute
-            ) b
-        ) c
+            SELECT -- This subquery truncates the block time to a minute and selects all the inbound tranfers of WETH tokens to the Fee GLP contract
+                date_trunc('minute', evt_block_time) AS minute,
+                ((value) / 1e18) AS transfer_value -- WETH 18dp
+            FROM {{ source('erc20_arbitrum', 'evt_transfer') }}
+            WHERE `to` = '0x4e971a87900b931ff39d1aad67697f49835400b6' -- Fee GLP contract
+            AND `contract_address` = '0x82af49447d8a07e3bd95bd0d56f35241523fbab1' -- WETH Arbitrum Smart Contract
+            ) a
+        GROUP BY a.minute
+        ) b
     )
 
 SELECT
-    a.minute AS minute,
-    COALESCE(b.fees_weth,0) AS fees_weth, -- Removes null values
-    last(c.weth_current_price, true) OVER (ORDER BY a.minute ASC) AS weth_current_price  -- extrapolation for differential is data table syncing
-FROM minute a    
-LEFT JOIN
-    (
-    SELECT
-        minute,
-        last(balance_weth, true) OVER (ORDER BY minute ASC) AS fees_weth
-    FROM fglp_balances 
-    ) b
-    ON a.minute = b.minute
-LEFT JOIN
+    x.minute,
+    COALESCE(x.fees_weth_generated,0) AS fees_weth_generated, -- Removes null values
+    COALESCE(x.fees_weth_cumulative,0) AS fees_weth_cumulative, -- Removes null values
+    x.weth_current_price
+FROM
     (
     SELECT
-        minute,
-        weth_current_price
-    FROM {{ref('gmx_arbitrum_glp_components')}}
-    ) c 
-    ON a.minute = c.minute
\ No newline at end of file
+        a.minute AS minute,
+        b.weth_transfer_value AS fees_weth_generated,
+        last(b.weth_cum_balance, true) OVER (ORDER BY a.minute ASC) AS fees_weth_cumulative, -- extrapolation
+        last(c.weth_current_price, true) OVER (ORDER BY a.minute ASC) AS weth_current_price -- extrapolation necessary for missing values arising from in data table syncing speed
+    FROM minute a    
+    LEFT JOIN
+        (
+        SELECT
+            minute,
+            weth_transfer_value,
+            weth_cum_balance
+        FROM fglp_balances 
+        ) b
+        ON a.minute = b.minute
+    LEFT JOIN
+        (
+        SELECT
+            minute,
+            weth_current_price
+        FROM {{ref('gmx_arbitrum_glp_components')}}
+        ) c 
+        ON a.minute = c.minute
+    ) x
```

Comment:
not gonna lie, I would love to see more meaningful subquery names. I think single letter subqueries are hard for other developers to interpret. I'll leave that for you to decide if you want to clean up.